src/App.tsx

```diff
@@ -1,25 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [visiblePerPage, setVisiblePerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
+  const pages: string[][] = Array.from(
+    { length: Math.ceil(items.length / visiblePerPage) },
+    (_, i) =>
+      items.slice(i * visiblePerPage, i * visiblePerPage + visiblePerPage),
+  );
+  const first = pages[currentPage].at(0);
+  const last = pages[currentPage].at(-1);
```

Comment:
Potential error: if `currentPage` is out of `pages` array bounds, it will cause an error because you're trying to access `.at(0)` and `.at(-1)` of `undefined`. It's better to add a check for this situation. For example, you can check if `pages[currentPage]` is defined before accessing its properties.